pkg/config/certs.go

```diff
@@ -41,9 +41,9 @@ func (c *Config) keyPath() string {
 	return filepath.Join(c.StateDir, keyFilename)
 }
 
-func (c *Config) getCertificate() (*tls.Certificate, error) {
+func (c *Config) CreateCertificate() (*tls.Certificate, error) {
```

Comment:
Maybe `GetOrCreateCertificate()`?

I agree that `getCertificate()` is less clear since it sounds like a read-only function but sometimes creates the files.